test/TestGen.hs

```diff
@@ -5,6 +5,10 @@ default(Maybe Bool,Int,Double,Maybe (Maybe Bool),Maybe (Maybe Char))
 tests :: IO ()
 tests = do
     let x ||| y = do t1 <- onceFork x; t2 <- onceFork y; t1; t2
+    testGen "withTempFile $ \\f1 -> withTempFile $ \\f2 -> fileEq f1 f2" $ withTempFile $ \f1 -> withTempFile $ \f2 -> fileEq f1 f2
+    testGen "withTempFile $ \\f -> copyFile \"test/TestGen.hs\" f >> fileEq \"test/TestGen.hs\" f" $ withTempFile $ \f -> copyFile "test/TestGen.hs" f >> fileEq "test/TestGen.hs" f
+    testGen "withTempFile $ \\f1 -> withTempFile $ \\f2 -> writeFile f1 \"a\" >> writeFile f2 \"a\" >> fileEq f1 f2" $ withTempFile $ \f1 -> withTempFile $ \f2 -> writeFile f1 "a" >> writeFile f2 "a" >> fileEq f1 f2
+    testGen "withTempFile $ \\f1 -> withTempFile $ \\f2 -> writeFile f1 \"a\" >> writeFile f2 \"b\" >> fmap not (fileEq f1 f2)" $ withTempFile $ \f1 -> withTempFile $ \f2 -> writeFile f1 "a" >> writeFile f2 "b" >> fmap not (fileEq f1 f2)
```

Comment:
I have thought about this in the past and discussed it with people. Certainly arguments both ways. However, I find it's easier to mark the files as generated (which I have now done!) and then have a test that generating doesn't change them (which I have always had). That way someone who wants to `git clone && cabal install` can do so, reducing the effort to get started. If there were _lots_ of generated files (e.g. a binding to Amazon generated from XML) then I'd not check them in, but at this level it seems worthwhile.